server/src/com/treode/server/LCSSyntaxAnalyzer.scala

```diff
@@ -0,0 +1,192 @@
+package com.treode.server
+import com.treode.twitter.finagle.http.{BadRequestException}
+import com.treode.async.misc.parseUnsignedLong
+import scala.util.{Success, Failure}
+import scala.util.parsing.combinator.lexical._
+import scala.util.parsing.combinator.token._
+import scala.util.parsing.input.CharArrayReader.EofCh
+import scala.collection.mutable.Queue
+import scala.collection.mutable.ArrayBuffer
+import scala.util.parsing.input.Position
+import scala.collection.Iterator
+import scala.collection.mutable.{HashMap, MutableList}
+
+
+
+class LCSSyntaxAnalyzer extends SchemaParserStdLexical {
+
+  trait Message {
+    def pos: Position
+    def message: String
+    override def toString = (message + "\n" + "at line " + pos.line + ", column " + pos.column + "\n" + pos.longString)
+  }
+  case class NotFoundMessage (pos: Position, message: String) extends Message
```

Comment:
It's odd to have `message` as a parameter. I see this case class is used for "Expected token...". So

```
case class ExpectedTokenMessage (pos: Position, token: String) {
  def message = s"Expected token $token"
}
```

Then when testing, you can write something like

```
assertResult (CompileFailure (ExpectedTokenMessage (Position (...), "token"))) (parse (...))
```